File: Java_Programmers/1단계/src/가운데글자가져오기.java
public class 가운데글자가져오기 {
    public String solution(String s) {
        int len = s.length();
        return s.substring((len-1) / 2, (len/2) + 1);
    }

    public static void main(String[] args) {
        가운데글자가져오기 sol = new 가운데글자가져오기();
        String s = "qwer";
        System.out.println(sol.solution(s));
    }
}

File: python/python_Programmers/1단계/가운데글자가져오기.py
# 내 풀이
def solution(s):
    answer = ''
    index = len(s)//2
    if len(s) % 2 == 0 : # 짝수
        answer += s[index-1]
        answer += s[index]
    else:
        answer += s[index]

    return answer

print(solution("abcde"))
print(solution("qwer"))

# 모범 풀이
def string_middle(str):
    a = len(str)
    if a % 2 == 0 :
        a = (a-2) / 2
    else :
        a = (a-1) / 2
    return str[int(a) : -int(a)]
# 아래는 테스트로 출력해 보기 위한 코드입니다.
print(string_middle("power"))